### ISSUE TITLE
Use `https` for the API calls or allow setting the protocol

### ISSUE THREAD

**Issue by ianjakobs** [microsoft/cognitive-services-speech-sdk-js#465]
First reported in #458, myself and a few others are still experiencing the following error despite the original issue having been closed by the author:

```
Unable to connect: StatusCode: 500, Protocol "http:" not supported. Expected "https:"
```

More details can be found there, but in short it seems that this SDK doesn't explicitly set a protocol for the API calls, causing the [Sentry JavaScript SDK](https://github.com/getsentry/sentry-javascript) to fall back to `http`.

Even though it looks like it's only us Sentry users that are running into this, it seems better to address it here since the Azure API requires `https`. Alternatively, it could help to be able to set the protocol explicitly through `sdk.SpeechConfig.fromEndpoint`, which is what I've tried to do as well.

For reference, here's the code that I'm using:

```typescript
const synthesizeSpeech = async (
    term: Translation['term'],
    language: Translation['to'],
): Promise<sdk.SpeechSynthesisResult> => {
    const speechConfig = sdk.SpeechConfig.fromSubscription(
        process.env.MICROSOFT_SPEECH_KEY,
        process.env.MICROSOFT_SPEECH_REGION,
    )

    speechConfig.speechSynthesisLanguage = language
    speechConfig.speechSynthesisVoiceName = LANGUAGES.find(({ code }) => code === language).voice
    speechConfig.speechSynthesisOutputFormat = sdk.SpeechSynthesisOutputFormat.Audio24Khz48KBitRateMonoMp3

    const synthesizer = new sdk.SpeechSynthesizer(speechConfig)

    return new Promise((resolve, reject) => synthesizer.speakTextAsync(
        term,
        result => {
            resolve(result)
            synthesizer.close()
        },
        error => {
            console.error(error)
            reject(error)
            synthesizer.close()
        }),
    )
}
```

**Comment by rhurey** [comment]
Some debugging this morning...

The tl;dr It looks like this is some friction between how the agent-base module decides if the connection is secure or not (It scans the callstack at the point of the request) and when Sentry is trying to determine what the protocol is.

**Comment by rhurey** [comment]
Looks like https://github.com/TooTallNate/node-agent-base/issues/61 is the most likely root cause.

Have a workaround I'll send a PR for.

**Comment by ianjakobs** [comment]
Yes, that seems to work! I installed from your branch with `npm i microsoft/cognitive-services-speech-sdk-js#rhurey/add_protocol` and followed the [recommended implementation for Next.js](https://docs.sentry.io/platforms/javascript/guides/nextjs/) (as I did before) and I'm now getting the expected response.

Thank you for the fast response and fix!

**Comment by rhurey** [comment]
Ok, the PR is in so I'm going to close this issue, lmk if you have problems.

**Comment by dargilco** [comment]
From the [1.20 release notes](https://docs.microsoft.com/azure/cognitive-services/speech-service/releasenotes?tabs=speech-sdk):
JavaScript: Add protocol to agent to mitigate bug found with Sentry integration (GitHub issue 465)
